src/core/meshcop/border_agent.cpp

```diff
@@ -776,13 +794,65 @@ void BorderAgent::Stop(void)
     Get<Tmf::SecureAgent>().Stop();
 
     mState        = kStateStopped;
+    mPskcChanged  = false;
     mUdpProxyPort = 0;
     LogInfo("Border Agent stopped");
 
 exit:
     return;
 }
 
+const char *BorderAgent::ReasonToString(RestartReason aReason)
+{
+    static const char *const kReasonStrings[] = {
+        "PskcChanged",               // (0) kRestartOnPskcChanged
+        "SecureSessionDisconnected", // (1) kRestartOnDisconnected
+        "EphemeralKeyRemoved",       // (2) kRestartOnEphemeralKeyRemoved
+
+    };
+
+    return kReasonStrings[aReason];
+}
+
+void BorderAgent::Restart(RestartReason aReason)
+{
+    uint16_t port = 0;
+
+#if OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
+    if (aReason == kRestartOnEphemeralKeyRemoved)
```

Comment:
Can consider uinsg `switch()`. Maybe something like this:
```c++
    switch (aReason)
    {
    case kRestartOnDisconnected:
        if (!mPskcChanged)
        {
            mState        = kStateStarted;
            mUdpProxyPort = 0;
            ExitNow();
        }

        OT_FALL_THROUGH;

    case kRestartOnPskcChanged:
        port = GetUdpPort();
        break;

#if OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
    case kRestartOnEphemeralKeyRemoved:
        port = mOldUdpPort;
        break;
#endif
    }
```

Reply:
did not use switch/case intentionally - with the switch approach snippet above, if pskc changes when the secure session is established via epskc, the switch approach here will reuse the ephemeral port, other than the original border agent port;  it can be mitigated by moving the `kRestartOnEphemeralKeyRemoved` up, however `case` conditions are generally considered equally, so here use `if` instead to process `kRestartOnEphemeralKeyRemoved` firstly.

Reply:
Not sure I follow. If I'm not mistaken, the `switch()` statement is doing the same thing as the `if` checks. Maybe I'm missing something?

I also noticed this comment:

```
  // `kRestartOnEphemeralKeyRemoved` should be processed before `kRestartOnDisconnected`
    // in case pskc has changed when there is a secure session established via epskc.
```

But `aReason` can only have one value at a time, right? And all the checks are in the form of `if (aReason == <something>)`, so only one of them will be executed?

Reply:
yes, after second thought, just updated the code to remove the reason and use variables mUsingEphemeralKey and mPskcChanged instead for proper process. Please take another review.